### PR TITLE
Populate code.gov zone

### PR DESCRIPTION
**Pull request by rossdakin**
Needs real zone_id.

### CODE REVIEW

File: terraform/code.gov.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+resource "aws_route53_record" "www" {
+  zone_id = "${aws_route53_zone.code_toplevel.zone_id}"
+  name = "code.gov."
+  type = "A"
+
+  alias {
+    name = "s3-website-us-east-1.amazonaws.com."
+    zone_id = "XXX"
```
Comment (rossdakin) [inline terraform/code.gov.tf]: @jezhumble can you help me figure out what to do here?

Unsure how to populate `zone_id` and how to specify the alias target (which I believe should be the ELB that @dlapiduz setup for us).

Thanks!!

**Comment by rossdakin** [comment]
CC @okamanda @mattbailey0

**Comment by rossdakin** [comment]
@jezhumble -- ELB to target is d3qovtf6opsrko.cloudfront.net -- is that helpful, or do you need some other identifier (resource ID, etc.)?

**Comment by dlapiduz** [comment]
@rossdakin I don't know much about route53 with terraform but I think you are looking for something like this:

```
variable "env_name" {}

resource "aws_route53_zone" "code_toplevel" {
   name = "code.gov"

   tags {
     Project = "dns"
   }
}

resource "aws_route53_record" "www" {
  zone_id = "${aws_route53_zone.code_toplevel.zone_id}"
  name = "www.code.gov."
  type = "CNAME"
  ttl = 60
  records = ["d3qovtf6opsrko.cloudfront.net."]
}

resource "aws_route53_record" "apex" {
  zone_id = "${aws_route53_zone.code_toplevel.zone_id}"
  name = "code.gov."
  type = "A"

  alias {
    name = "www.code.gov"
    zone_id = "${aws_route53_zone.code_toplevel.zone_id}"
    evaluate_target_health = false
  }
}

output "code_ns" {
    value="${aws_route53_zone.code_toplevel.name_servers}"
}
```

**Comment by rossdakin** [comment]
Sorry @jezhumble, I'm dumb -- `d3qovtf6opsrko.cloudfront.net` is a cloudfront distro, not an ELB.

Can we just CNAME to it like in Diego's example?

**Comment by jezhumble** [comment]
It's a sneaky idea, but I tried it and it doesn't work (Terraform reports "aws_route53_record.www: resource repeated multiple times"). Route 53 doesn't let you use a CNAME for an apex domain, and aliasing to a CNAME is still using a CNAME fundamentally.

In order to get this to work, I need the zone id for the cloudfront distro (d3qovtf6opsrko.cloudfront.net). If someone can even tell me the AWS account the cloudfront distro is set up in, that will suffice.

In the meantime, I've created an empty zone file. The NS records to put in dotgov.gov are as follows:

ns-1378.awsdns-44.org
ns-1773.awsdns-29.co.uk
ns-418.awsdns-52.com
ns-564.awsdns-06.net

**Comment by dlapiduz** [comment]
(Just found out a cool tip, a cloudfront zone id is always `Z2FDTNDATAQYW2` regardless of the account)

@jezhumble  can you try:

```
resource "aws_route53_record" "apex" {
  zone_id = "${aws_route53_zone.code_toplevel.zone_id}"
  name = "code.gov."
  type = "A"

  alias {
    name = "d3qovtf6opsrko.cloudfront.net."
    zone_id = "Z2FDTNDATAQYW2"
    evaluate_target_health = false
  }
}
```

I thought we had the alias to www on cloud.gov but we have the alias straight to cloudfront.

**Comment by rossdakin** [comment]
Updated with what (I think) the proposal is.

@dlapiduz @jezhumble -- does this look right?

Thanks again!

**Comment by rossdakin** [comment]
Huge thanks to @dlapiduz and @jezhumble for your help on this!

**Pull request merged**